src/test/java/edu/ucsb/cs156/gauchoride/controllers/RideControllerTests.java

```diff
@@ -0,0 +1,246 @@
+         String responseString = response.getResponse().getContentAsString();
+         assertEquals(expectedJson, responseString);
+     }
+ 
+     @WithMockUser(roles = { "RIDER" })
+     @Test
+     public void logged_in_rider_cannot_post_for_others() throws Exception {
+         User currentUser = currentUserService.getCurrentUser().getUser();
+         LocalTime startT = LocalTime.parse("14:00:00");
+         LocalTime stopT = LocalTime.parse("15:15:00");
+         User otherUser = User.builder().id(33L).build();
+         User driver = User.builder().id(10L).driver(true).build();
+         Ride ride1 = Ride.builder()
+         .day("Tuesday")
+         .rider(otherUser)
+         .driver(driver)
+         .course("CMPSC 156")
+         .timeStart(startT)
+         .timeStop(stopT)
+         .building("South Hall")
+         .room("1431")
+         .pickUp("Home")
+         .phoneNumber(8001230101L)
+         .build();
+ 
+         Ride ride2 = Ride.builder()
+         .day("Tuesday")
+         .rider(currentUser)
+         .driver(driver)
+         .course("CMPSC 156")
+         .timeStart(startT)
+         .timeStop(stopT)
+         .building("South Hall")
+         .room("1431")
+         .pickUp("Home")
+         .phoneNumber(8001230101L)
+         .build();
+         
+         String requestBody = mapper.writeValueAsString(ride1);
+         when(rideRepository.save(eq(ride2))).thenReturn(ride2);
+ 
+         // act
+         MvcResult response = mockMvc.perform(post("/api/rides/post")
+         .contentType(MediaType.APPLICATION_JSON)
+         .characterEncoding("utf-8")
+         .content(requestBody).with(csrf())).andExpect(status().isOk()).andReturn();
+ 
+         // assert
+         verify(rideRepository, times(1)).save(ride2);
+         String expectedJson = mapper.writeValueAsString(ride2);
+         String responseString = response.getResponse().getContentAsString();
+         assertEquals(expectedJson, responseString);
+     }
+ 
+     @WithMockUser(roles = { "RIDER" })
+     @Test
+     public void logged_in_rider_cannot_post_a_new_ride_request_with_no_driver() throws Exception {
+         User currentUser = currentUserService.getCurrentUser().getUser();
+         LocalTime startT = LocalTime.parse("14:00:00");
+         LocalTime stopT = LocalTime.parse("15:15:00");
+         User otherUser = User.builder().id(33L).build();
+         User driver = User.builder().id(10L).driver(false).build();
+         Ride ride = Ride.builder()
+         .day("Tuesday")
+         .rider(currentUser)
+         .driver(driver)
+         .course("CMPSC 156")
+         .timeStart(startT)
+         .timeStop(stopT)
+         .building("South Hall")
+         .room("1431")
+         .pickUp("Home")
+         .phoneNumber(8001230101L)
+         .build();
+ 
+         String requestBody = mapper.writeValueAsString(ride);
+         when(rideRepository.save(eq(ride))).thenReturn(ride);
+ 
+         // Act
+         assertThatThrownBy(() -> {

```

Comment:
```suggestion
         mockMvc.perform(post("/api/rides/post")
        .contentType(MediaType.APPLICATION_JSON)
        .characterEncoding("utf-8")
        .content(requestBody).with(csrf()))
        .andExpect(status().isBadRequest());

```

Reply:
this is used to fix your test since now if the driver in the ride object doesn't have the driver field set to true then it will return a bad request instead. So you can remove the entire assertThatThrownBy block, and add the code above. Also, remove any unused variables or import statements in your PR if you see any.

Reply:
Addressed